modules/dnn/src/layers/elementwise_layers.cpp

```diff
@@ -1410,6 +1410,7 @@ struct ExpFunctor : public BaseFunctor
         : base(base_), scale(scale_), shift(shift_)
     {
         CV_Check(base, base == -1.f || base > 0.f, "Unsupported 'base' value");
+        finalize(); // init normScale and normShift
```

Comment:
It make sense to move finalize "body" here instead (and keep finalize empty).

Reply:
I'm not sure, for some reason functors were designed to implement layer's `finalize` method and all data members of this class are public, so they can be changed externally and then _finalized_ to calculate norm parameters.

**cc** @SamFC10 , @l-bat

Reply:
Only `ExpLayer::create()` (below) knows about this class.
Base class assumes "updates" through `tryFuse()` callback, but it is not implemented here. Perhaps even `finalize()` with empty body is not needed.

Reply:
Isn't it meant to be called here?https://github.com/opencv/opencv/blob/7328600a7a5e3a2a925c2581c9c5149a1ccc7065/modules/dnn/src/layers/elementwise_layers.cpp#L129-L132

Reply:
There is already empty implementation in the base class:

https://github.com/opencv/opencv/blob/7328600a7a5e3a2a925c2581c9c5149a1ccc7065/modules/dnn/src/layers/elementwise_layers.cpp#L286

Reply:
OK, updated.

Reply:
My initial thought was to initialize the norm fields here and not have `finalize()` at all like @alalek suggests. I added it to `finalize()` instead as the PowerLayer above has a similar `finalize()` method and thought initializing norm fields in finalize() instead of the constructor "shouldn't" really affect the layer's implementation as finalize() is called before forward() anyway. Now I think its better to shift finalize() body here and remove it.